progenitor/src/main.rs

```diff
@@ -171,7 +172,18 @@ pub fn load_api<P>(p: P) -> Result<OpenAPI>
 where
     P: AsRef<Path>,
 {
-    let f = File::open(p)?;
-    let api = serde_json::from_reader(f)?;
+    let mut f = File::open(p)?;
+
+    let mut buf = [b' '];
+    while buf[0].is_ascii_whitespace() {
```

Comment:
this doesn't seem right: a json file could start with whitespace, couldn't it?

Reply:
This loops ignores all leading whitespace in the file, so that the `if` below is checking whether the first non-whitespace character is `{`.

Reply:
right! fair enough. Sounds like you prefer this to the options I mentioned?

Reply:
Well, IMO we need some auto-detection to handle https://github.com/oxidecomputer/progenitor/issues/228 in order to still do the right thing if the content-type isnt recognised , and often raw json & yaml files have strange content-types provided by the server.

Reply:
yes, agreed. I suggested that we try to parse as json and then, failing that, parse as yaml. Would that not suffice?

Reply:
Yea, I like that, and will do that next time I revisit this PR.